src/openms/source/FORMAT/Base64.cpp

```diff
@@ -1,123 +1,107 @@
+    data = (!OPENMS_IS_BIG_ENDIAN ? data.shuffle_epi8(shuffle_mask_1_) : data.shuffle_epi8(shuffle_mask_big_endian_));
 
-  // difference between base64 encoding and ascii encoding, used to cast from base64 binaries to characters
+    data = (data.srli_epi32(2) & mask1_) | (data.srli_epi32(4) & mask2_) |
+           (data.srli_epi32(6) & mask3_) | (data & mask4_);
 
-  const simde__m128i difference_A_ = simde_mm_set1_epi8('A');
-  const simde__m128i difference_a_ = simde_mm_set1_epi8('a' - 26);
-  const simde__m128i difference_0_ = simde_mm_set1_epi8('0' - 52);
-  const simde__m128i difference_plus_ = simde_mm_set1_epi8('+');
-  const simde__m128i difference_slash_ = simde_mm_set1_epi8('/');
+    if (!OPENMS_IS_BIG_ENDIAN)
+      data = data.shuffle_epi8(shuffle_mask_2_);
 
-  const simde__m128i shuffle_mask_1_ = simde_mm_setr_epi8(2, 2, 1, 0, 5, 5, 4, 3, 8, 8, 7, 6, 11, 11, 10, 9);
-  const simde__m128i shuffle_mask_2_ = simde_mm_setr_epi8(3, 2, 1, 0, 7, 6, 5, 4, 11, 10, 9, 8, 15, 14, 13, 12);
+    Vec128 capital_mask = Vec128::cmplt_epi8(data, Vec128(simde_mm_set1_epi8(26)));
+    Vec128 all_mask = capital_mask;
 
-  const simde__m128i shuffle_mask_big_endian_ = simde_mm_setr_epi8(0, 1, 2, 2, 3, 4, 5, 5, 6, 7, 8, 8, 9, 10, 11, 11);
-  // second shuffle doesnt need to happen
+    Vec128 lower_case_mask = Vec128::andnot(all_mask, Vec128::cmplt_epi8(data, Vec128(simde_mm_set1_epi8(52))));
+    all_mask |= lower_case_mask;
 
-  // decoding shuffle masks:
-  // shuffle_mask_2 gets used
-  const simde__m128i shuffle_mask_d_2_ = simde_mm_setr_epi8(3, 2, 1, 7, 6, 5, 11, 10, 9, 15, 14, 13, 0, 4, 8, 12);
+    Vec128 number_mask = Vec128::andnot(all_mask, Vec128::cmplt_epi8(data, Vec128(simde_mm_set1_epi8(62))));
+    all_mask |= number_mask;
 
-  /// Encode the first 12 bytes of a 128 bit simde integer type to base64
-  void registerEncoder_(simde__m128i& data)
-  {
-    if constexpr (!OPENMS_IS_BIG_ENDIAN)
-    {
-      data = simde_mm_shuffle_epi8(data, shuffle_mask_1_);
-      // by shuffling every 3 8bit ASCII Letters now take up 4 bytes, "ABC" gets shuffled to "CCBA" to match the 4 bytes of the Base64 Encoding, and deal with little Endianness.
-    }
-    else
-    {
-      data = simde_mm_shuffle_epi8(data, shuffle_mask_big_endian_);
-    }
-    // shifting and masking data, so now every 6 bit of the ASCII encoding have their own byte.
-    // shifting over 32 bits takes endianness into account, which needs to be accounted for when shuffeling
-    data = (simde_mm_srli_epi32(data, 2) & mask1_) | (simde_mm_srli_epi32(data, 4) & mask2_) | (simde_mm_srli_epi32(data, 6) & mask3_) | (data & mask4_);
+    Vec128 plus_mask = Vec128::andnot(all_mask, Vec128::cmplt_epi8(data, Vec128(simde_mm_set1_epi8(63))));
+    all_mask |= plus_mask;
 
-    if constexpr (!OPENMS_IS_BIG_ENDIAN) // otherwise the data already is ordered correctly
-    {
-      data = simde_mm_shuffle_epi8(data, shuffle_mask_2_);
-    }
+    Vec128 slash_mask = all_mask;
```

Comment:
no need to copy and rename.
the code before used
`simde__m128i& slash_negative_mask = all_mask;`
Please do the same, unless you can prove that this makes the code faster (benchmark).